website/content/60-advanced/601-emr-integration/6013-before-you-access-data/60132-update-callback.md

```diff
@@ -19,5 +19,28 @@ autoNav = true
     <li>Scroll down to the bottom of the page and click on <b>Next</b> and then click on <b>Finish</b> on the Feedback tab.</li>
+  </ol>
+  <h4>For AD FS </h4>
+  <ol>
+    <li>Login into Windows Server 2019 you connected in the previous chapter, Open PowerShell and execute the following commands, make sure to replace <b style="color:red">account-id</b> with your AWS AccountId and <b style="color:red">public-dns</b> with EMR cluster's Master Public DNS, you can find the value in CloudFormation stack output <b>EMRMasterNodeDNS</b>, use some Text editor to make these changes. <br/><br/> Here we are downloading Knox public certificate from S3 bucket ( As part of Step function inside EMR cluster creation we have extracted Knox public certificate and uploaded to your S3 bucket) and updating Relying Party Trust with Knox certificate, Claim Rules as well as SAML endpoint URL with Amazon EMR cluster's master DNS.
+    <pre>
+Copy-S3Object -BucketName lf-workshop-<b style="color:red">account-id</b> -Key metadata/knox.cer -LocalFile C:\\cfn\knox.cer
+
+$KnoxCert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
+
+$KnoxCert.Import("c:/cfn/knox.cer")
+
+$EP = New-AdfsSamlEndpoint -Binding "POST" -Protocol "SAMLAssertionConsumer" -Uri "https://<b style="color:red">public-dns</b>:8442/gateway/knoxsso/api/v1/websso?pac4jCallback=true&client_name=SAML2Client“
+
+$RuleSetAll = New-AdfsClaimRuleSet -ClaimRule ('c:[Type == "http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname"]
+ => issue(Type = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier", Issuer = c.Issuer, OriginalIssuer = c.OriginalIssuer, Value = c.Value, ValueType = c.ValueType, Properties["http://schemas.xmlsoap.org/ws/2005/05/identity/claimproperties/format"] = "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent");','c:[Type == "http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname", Issuer == "AD AUTHORITY"]
+  => issue(store = "Active Directory", types = ("https://aws.amazon.com/SAML/Attributes/RoleSessionName"), query = ";sAMAccountName;{0}", param = c.Value);','c:[Type == "http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname", Issuer == "AD AUTHORITY"]
+   => issue(store = "Active Directory", types = ("https://lakeformation.amazon.com/SAML/Attributes/Username"), query = ";sAMAccountName;{0}", param = c.Value);','=> issue(Type = "https://aws.amazon.com/SAML/Attributes/Role", Value = "arn:aws:iam::<b style="color:red">account-id</b>:role/LF-SAML-Role,arn:aws:iam::<b style="color:red">account-id</b>:saml-provider/ADFSSAMLProvider");')
+
+Set-AdfsRelyingPartyTrust -TargetIdentifier urn:amazon:webservices -RequestSigningCertificate $KnoxCert -SamlEndpoint $EP  -IssuanceTransformRules $RuleSetAll.ClaimRulesString
+    </pre><img src="/images/adfs-updatecallbackurl.png" title="Update ADFS Callback URL" style="margin:15px 0px; border:1px solid black"/> </li>
+
```

Comment:
Hard to see anything on the picture, either increase size or remove it

Reply:
Increased the size by taking with bigger fonts in windows.